atlas/resource_atlas_schema_test.go

```diff
@@ -106,3 +113,154 @@ func TestAccAtlasDatabase(t *testing.T) {
 		},
 	})
 }
+
+const testAccValidSqliteSchema = `
+resource "atlas_schema" "testdb" {
+  hcl = <<-EOT
+	table "orders" {
+		schema = schema.main
+		column "id" {
+			null = true
+			type = int
+		}
+	}
+	schema "main" {
+	}
+	EOT
+  url = "sqlite://database.sqlite3?cache=shared"
+}
+`
+
+const testAccInvalidSqliteSchema = `
+resource "atlas_schema" "testdb" {
+  hcl = <<-EOT
+	table "orders {
+		schema = schema.main
+		column "id" {
+			null = true
+			type = int
+		}
+	}
+	schema "main" {
+	}
+	EOT
+  url = "sqlite://database.sqlite3?cache=shared"
+}
+`
+
+func TestAccInvalidSchemaReturnsError(t *testing.T) {
+	t.Cleanup(func() {
+		os.Remove("database.sqlite3")
+	})
+	resource.Test(t, resource.TestCase{
+		Providers: map[string]*schema.Provider{
+			"atlas": Provider(),
+		},
+		Steps: []resource.TestStep{
+			{
+				Config:             testAccValidSqliteSchema,
+				ExpectNonEmptyPlan: true,
+			},
+			{
+				Config:      testAccInvalidSqliteSchema,
+				ExpectError: regexp.MustCompile("schemahcl: failed decoding"),
+			},
+		},
+	})
+
+	url := "sqlite://database.sqlite3?cache=shared"
+	cli, err := sqlclient.Open(context.Background(), url)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	realm, err := cli.InspectRealm(context.Background(), nil)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	tbl, ok := realm.Schemas[0].Table("orders")
+	if !ok {
+		t.Error("expected database to have table \"orders\"")
+		return
+	}
+	if _, ok := tbl.Column("id"); !ok {
+		t.Error(fmt.Errorf("expected database to have table \"orders\" but got: %s", realm.Schemas[0].Tables[0].Name))
+		return
+	}
+}
+
+const createTableStmt = `create table type_table
+(
+    tBit                        bit(10)              default 4                                              null,
+    tInt                        int(10)              default 4                                               not null,
+    tTinyInt                    tinyint(10)          default 8                                                   null
+) CHARSET = utf8mb4 COLLATE utf8mb4_0900_ai_ci;`
+
+var testAccSanity = fmt.Sprintf(`
+data "atlas_schema" "sanity" {
+  dev_db_url = "%s"
+  src = <<-EOT
+	table "type_table" {
+		schema  = schema.test
+		charset = "utf8mb4"
+		collate = "utf8mb4_0900_ai_ci"
+		column "tInt" {
+			null    = false
+			type    = int
+			default = 4
+		}
+	}
+	schema "test" {
+		charset = "utf8mb4"
+		collate = "utf8mb4_0900_ai_ci"
+	}
+	EOT
+}
+resource "atlas_schema" "testdb" {
+  hcl = data.atlas_schema.sanity.hcl
+  url = "%s"
+}
+`, MYSQL_DEV_URL, MYSQL_URL)
+
+const sanityState = `table "type_table" {
+  schema = schema.test
+  column "tInt" {
+    null    = false
+    type    = int
+    default = 4
+  }
+}
+schema "test" {
+  charset = "utf8mb4"
+  collate = "utf8mb4_0900_ai_ci"
+}
+`
+
+func TestAccRemoveColumns(t *testing.T) {
```

Comment:
What is "Acc"?

Reply:
https://www.terraform.io/plugin/sdkv2/testing/acceptance-tests#test-files